lib/using_restricted_array.rb

```diff
@@ -6,56 +6,132 @@
 
 # Calculates the length of the restricted array. All values are integers.
 # The restricted_array is terminated by 'nil' i.e. array[length] = nil
-# Time complexity: ?
-# Space complexity: ?
+# Time complexity: O(n)
+# Space complexity: O(1)
 def length(array)
-  raise NotImplementedError
+  i = 0
+  length = 0
+  
+  while array[i] != nil
+    length += 1
+    i += 1
+  end
+  
+  return length
 end
 
 # Prints each integer values in the array
-# Time complexity: ?
-# Space complexity: ?
+# Time complexity: O(n)
+# Space complexity: O(n)
```

Comment:
Since you don't use more data the larger n gets, this method has space complexity O(1)